### PR TITLE
Fix #3250 - Makes form wizard the default

### LINKED ISSUE #3250: Actually Enable form v2 for 100% of people
We enabled form v2 for 100% of new users... but it's not really enabled, depending on DNT and *possibly* logged in status. 

https://github.com/webcompat/webcompat.com/blob/d1c3b122fac11a2c6cc838c67cc358a23d5df8a6/webcompat/helpers.py#L706-L733

I think we should remove the DNT restriction -- if everyone has it on, it's not tracking. And we're not storing or sharing that info.

### PR DESCRIPTION
**Pull request by karlcow**
This PR fixes issue #3250

## Proposed PR background

My feeling is that if we are out of the experimental phase, we probably need to just not go through the AB code at all, and all of it becomes a lot simpler.

We can keep in place the code for future AB experiments and open a new issue for removing the old form code and tests once @ksy36 has finished refactoring the rest of the code for the new form.

So on this hypothesis, let's create a new PR. I expect tests to break badly. That's an Achilles' heel of our current setup. The AB experiment was an experiment at the beginning. Never let an experiment grows without the proper setup. We need to fix it.

```
(env) ~/code/webcompat.com % pytest
============================= test session starts ==============================
platform darwin -- Python 3.7.4, pytest-5.3.5, py-1.8.1, pluggy-0.13.1
rootdir: /Users/karl/code/webcompat.com
collected 157 items

tests/unit/test_api_urls.py ...........                                  [  7%]
tests/unit/test_config.py ..                                             [  8%]
tests/unit/test_console_logs.py .....                                    [ 11%]
tests/unit/test_form.py .................                                [ 22%]
tests/unit/test_helpers.py ............................                  [ 40%]
tests/unit/test_http_caching.py ...                                      [ 42%]
tests/unit/test_issues.py ......                                         [ 45%]
tests/unit/test_rendering.py ......                                      [ 49%]
tests/unit/test_tools_changelog.py ..                                    [ 50%]
tests/unit/test_topsites.py ...                                          [ 52%]
tests/unit/test_uploads.py ...                                           [ 54%]
tests/unit/test_urls.py .................................                [ 75%]
tests/unit/test_webhook.py ......................................        [100%]

============================= 157 passed in 21.28s =============================
```

Unit testing seems to work. There was a simple fix. The big breakage should happen on the functional tests side. Let's create a PR and see how much CircleCI is not happy about it.

### CODE REVIEW

**Comment by karlcow** [comment]
```
No unit test coverage for chrome 75.0.3770.100 on Linux
chrome 75.0.3770.100 on Linux: 99 passed, 2 failed
TOTAL: tested 2 platforms, 198 passed, 4 failed
Stopping
Stopped
something bad happened inside intern.run()
{ Error: One or more tests failed
    at /home/circleci/repo/node_modules/intern/lib/executors/Executor.js:386:45
    at /home/circleci/repo/node_modules/@theintern/common/index.js:16:7174
    at process._tickCallback (internal/process/next_tick.js:68:7) reported: true }

Exited with code exit status 1
```


This test

```
127.0.0.1 - - [24/Mar/2020 02:28:50] "GET /issues/new HTTP/1.1" 200 -

× firefox on linux 4.15.0-1052-aws - Reporting (non-auth) - Description validation (1.113s)

    Error: [POST http://localhost:4444/wd/hub/session/97201fda-3d8f-4f01-8b55-252780618c70/element/b426a6ea-a2ab-4fcd-823f-bb9fbe157fc1/value / {"value":["b","u","g"," ","d","e","s","c","r","i","p","t","i","o","n"]}] Element <input id="description" class="form-field text-field required" name="description" type="hidden"> is not reachable by keyboard

    Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'

    System info: host: '7dea615d62c3', ip: '192.168.144.3', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-1052-aws', java.version: '11.0.3'

    Driver info: driver.version: unknown

      at Test.Description validation [as test] @ tests/functional/reporting-non-auth.js:148:12

      @ src/lib/Test.ts:266:51

```

and this test

```
127.0.0.1 - - [24/Mar/2020 02:28:52] "GET /issues/new HTTP/1.1" 200 -

× firefox on linux 4.15.0-1052-aws - Reporting (non-auth) - Submits are enabled (0.385s)

    Error: [POST http://localhost:4444/wd/hub/session/97201fda-3d8f-4f01-8b55-252780618c70/element/c729c5a0-fe29-4119-9ee9-42873bce6c36/value / {"value":["t","e","s","t"," ","f","o","r"," ","d","e","s","k","t","o","p"," ","s","i","t","e"]}] Element <input id="description" class="form-field text-field required" name="description" type="hidden"> is not reachable by keyboard

    Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'

    System info: host: '7dea615d62c3', ip: '192.168.144.3', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-1052-aws', java.version: '11.0.3'

    Driver info: driver.version: unknown

      at Test.Submits are enabled [as test] @ tests/functional/reporting-non-auth.js:226:12

      @ src/lib/Test.ts:266:51

```

Let's see if we can fix those two.

**Comment by karlcow** [comment]
ok fixed one test… maybe. circleci will tell us if we are off.

**Comment by karlcow** [comment]
Fixing the test will be painful. 
Also why this is not working anymore.

It dies silently

node ./tests/functional/_intern.js "--functionalSuites=tests/functional/reporting-non-auth.js"

**Comment by karlcow** [comment]
🤔 puzzling

```
    NoSuchElement: [POST http://localhost:4444/wd/hub/session/27d4733c1552221fdb27c20e4cd4f712/element / {"using":"css selector","value":".step-1"}] 
no such element: Unable to locate element: {"method":"css selector","selector":".step-1"}
```

what do i do wrong?

**Comment by karlcow** [comment]
I wish i could run leadfoot inside the browser console.

**Comment by karlcow** [comment]
oh wait the new form is already tested in `/tests/functional/reporting-issue-wizard-non-auth.js`
ok let's remove the other one. and remove the cookie and if it fails, I guess it could be because of caching and that would explain why it was failing.

**Comment by karlcow** [comment]
ok this is more logical all tests are failing for the wizard form. so that means that the files are being cached. 

```
× firefox on linux 4.15.0-1052-aws - Reporting with wizard - Wizard stepper - scenario 2 (10.254s)

    NoSuchElement: [POST http://localhost:4444/wd/hub/session/d1d9060f-fbfe-4f81-8c89-2314ac4dcb0d/element / {"using":"css selector","value":".button.step-1"}] Unable to locate element: .button.step-1

    For documentation on this error, please visit: https://www.seleniumhq.org/exceptions/no_such_element.html

    Build info: version: '3.141.59', revision: 'e82be7d358', time: '2018-11-14T08:25:53'

    System info: host: '66b5d67c4eeb', ip: '172.22.0.3', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-1052-aws', java.version: '11.0.3'

    Driver info: driver.version: unknown

      at Test.Wizard stepper - scenario 2 [as test] @ tests/functional/reporting-issue-wizard-non-auth.js:164:12

      @ src/lib/Test.ts:266:51

```

**Comment by karlcow** [comment]
I exhausted github through circleci for a little while. Let's wait.

**Comment by ksy36** [comment]
I've checked out the branch, looks like it's not working because this condition is still in the templates (layout.html, new-issue.html):

`{%- if ab_active('exp') == 'form-v2' %}`

**Comment by ksy36** [comment]
Also this should be removed? Probably no point of adding the label if all reports will be going through the new form

https://github.com/webcompat/webcompat.com/blob/fdf8bef0ec72fdb6e94ea26cee77bf557ccf7e0a/webcompat/helpers.py#L763-L768

File: webcompat/views.py [hunk 1]
```diff
@@ -265,14 +265,8 @@ def create_issue():
...
-
-        if ab_active('exp') == 'form-v2':
-            bug_form = get_form(form_data, form=FormWizard)
-            pagetitle = "New Issue |"
-        else:
-            bug_form = get_form(form_data)
-            pagetitle = "New Issue"
-
+        bug_form = get_form(form_data, form=FormWizard)
+        pagetitle = "New Issue"
```
Comment (ksy36) [inline webcompat/views.py]: Should the title still remain as `pagetitle = "New Issue |"` here? I think this was added for ga tracking to differentiate the reports from old and new form (not sure if this being tracked still though)

**Comment by karlcow** [comment]
* yes the labeling should be removed. Good point. 
* `{%- if ab_active('exp') == 'form-v2' %}` and still manually I receive the right thing, but yes I will check this too. Good call. Thanks.

**Comment by karlcow** [comment]
ok this is much better now that I have fixed my issue with #3255 and a couple of other things.

Capture d’écran 2020-03-25 à 13 06 00

**Comment by karlcow** [comment]
These are the tests failing

* [ ] Image Uploads (non-auth) - Selecting an invalid image after a valid one 
* [ ] Image Uploads (non-auth) - Image extension validation
* [ ] Reporting (auth) - Report button shows name
* [ ] Reporting through postMessage - postMessaged object

**Comment by karlcow** [comment]
Capture d’écran 2020-03-25 à 16 46 00


Only two tests to fix, and then we are good to rebase a bit the things and ride. :)

**Comment by karlcow** [comment]
Capture d’écran 2020-03-26 à 12 00 44

ok all green locally. not let's rebase this. :)

**Comment by karlcow** [comment]
@ksy36 This is request for another review. the code changed quite a bit and i modified things with your recommendations.

@miketaylr I decided to make the wizard form the default. 

also each commit as an extended description explaining the line of thoughts.

File: config/__init__.py [hunk 2]
```diff
@@ -226,25 +225,23 @@ def get_variation(variation_key, variations_dict, defaults_dict):
 
 # Get AB experiment variation values from the environement.
 AB_VARIATIONS = {
-    'FORM_V1_VARIATION': os.environ.get('FORM_V1_VARIATION', '0 100'),
-    'FORM_V2_VARIATION': os.environ.get('FORM_V2_VARIATION', '100 100'),
+    'V1_VARIATION': os.environ.get('V1_VARIATION', '0 100'),
```
Comment (miketaylr) [inline config/__init__.py]: @karlcow can you make sure to update the environment variables on the server to match this?

File: webcompat/views.py (hunk 1 above)
Comment (miketaylr) [reply webcompat/views.py]: I asked Christos, and he said it's fine to remove it.

File: config/__init__.py (hunk 2 above)
Comment (miketaylr) [reply config/__init__.py]: (i did this)

**Comment by miketaylr** [comment]
I think we're good to merge and deploy (to staging first).

**Pull request merged**